File: enterprise-crm-frontend/src/app/models/Campaign.ts
import { NotificationType } from './NotificationType';

export class Campaign {
    campaignID?: number;
    name!: string;
    startDate!: Date;
    endDate!: Date;
    type!: NotificationType;
    customerInteractions?: number;
    trackingUrl?: string;
    status?: string;
}

File: CRM-Microservices-backend/marketing-automation/src/main/java/com/crm/entities/Campaign.java
package com.crm.entities;
import com.crm.enums.Type; 
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
@Setter
@Getter
@Entity
@Table(name = "campaign")
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Campaign {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "campaign_id",nullable = false)
    private Long campaignID;
    @Column(unique=true) 
    private String name;
    private LocalDate startDate;
    private LocalDate endDate;
    @Enumerated(EnumType.STRING)
    @Column(name="type")
    private Type type;
    private int customerInteractions;
    @Column(name="tracking_url")
    private String trackingUrl;

}